plugins/zns/zns.c

```diff
@@ -572,6 +572,18 @@ static int report_zones(int argc, char **argv, struct command *cmd, struct plugi
 	__u32 report_size;
 	void *report;
 	bool huge = false;
+	struct nvme_zone_report *buff;
+
+	unsigned int nr_zones_chunks = 128,
```

Comment:
Have you considered using the MDTS value (in the Identify Controller data structure) to determine the maximum transfer size?

If not, I would recommend using that as the transfer size or maybe something like 32-64KB (good tradeoff wrt to I/O submission/completion, and data to be sent/represented)

Reply:
@MatiasBjorling  I can increase the transfer size to 64KB.  Using the MDTS value could still fail with some drives.  My experience with linux is that it starts hitting intermittent buffer mapping and allocation failures with buffers larger then 500k.   64K seems to be a good compromise.

Reply:
I increased the buffer/transfer size to 64K (1000 entries at 64 bytes per entry).

Reply:
You'll need 1024 entries if you want to hit 64 KiB :)

Reply:
@MatiasBjorling You're correct and my comment above is wrong.  I did use 1024 in the code change so we should be good there.